common/reactive-grpc-common/src/main/java/com/salesforce/reactivegrpc/common/AbstractSubscriberAndProducer.java

```diff
@@ -276,14 +266,18 @@ void drainSync() {
                 if (v == null) {
                     try {
                         a.onCompleted();
-                    } catch (Throwable throwable) {
-                        cancel();
-                        a.onError(prepareError(throwable));
```

Comment:
Why was this guard deleted?

Reply:
Due to java docs, this any further interactions with Observer are forbidden -> https://github.com/grpc/grpc-java/blob/master/stub/src/main/java/io/grpc/stub/StreamObserver.java#L75